pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -604,6 +604,12 @@ func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, _ v1.PodStatus, podStat
 		podIP = podStatus.IP
 	}
 
+	rawHostIP, err := m.runtimeHelper.GetHostIP()
+	hostIP := rawHostIP.String()
```

Comment:
IIUC, `pod.Status.HostIP` is from [here: getHostIPAnyWay](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go#L1176). Node address is from as is mentioned in https://github.com/kubernetes/kubernetes/pull/43666#discussion_r108087166 . So IMO, if `nodeIP` in your link is set by kubelet flag `--node-ip`, then it will match `pod.Status.HostIP`.